src/service/viewport/viewport-binding-ios-embed-wrapper.js

```diff
@@ -216,6 +216,15 @@ export class ViewportBindingIosEmbedWrapper_ {
     return this.wrapper_./*OK*/scrollHeight;
   }
 
+  /** @override */
+  getContentHeight() {
+    // The reparented body inside wrapper will have the correct content height.
+    // Body is overflow: hidden so that the scrollHeight include the margins of
+    // body's first and last child.
+    const rect = this.win.document.body./*OK*/getBoundingClientRect();
+    return rect.height + rect.top + this.getScrollTop();
```

Comment:
Can you confirm what this does? Why would scroll top be needed here?

Reply:
Updated the comment to clarify.

I couldn't think of a better way to get the scrollTop back out of the DOM - offsetTop for the body is zero, and reading paddingTop out of the html element's style is messy. Alternatively could store paddingTop when setPaddingTop() is called and add it on here? This method is more generic though - would work with e.g. marginTop on body instead of paddingTop on html.